sql/src/main/java/io/crate/planner/operators/StatementClassifier.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.operators;
+
+import io.crate.planner.Plan;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Utility to classify SQL statements based on their {@link Plan}.
+ *
+ * A classification consists of the {@link io.crate.planner.Plan.StatementType} that is provided by the {@link Plan}
+ * implementation, such as DDL, SELECT, INSERT, UPDATE, DELETE, etc., and in case of a SELECT statement, additional
+ * labels. These labels are a set of {@link LogicalPlan}s that the statement plan consists of, for example:
+ *
+ * <pre>
+ *     SELECT * FROM users ORDER BY age DESC;
+ * </pre>
+ *
+ * Would result in a classification of:
+ *
+ * <pre>
+ *     type = SELECT
+ *     labels = [FetchOrEval, Collect, Order]
+ * </pre>
+ */
+public class StatementClassifier {
```

Comment:
Maybe make it final to make it clear that this is not designed to be sub-classed and/or add a private constructor to make it impossible to instantiate